src/System.Text.RegularExpressions/tests/Performance/Perf.Regex.Cache.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.Xunit.Performance;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class Perf_Regex_Cache
+    {
+        private const int N = 50_000;
+        private const int UniqueRegsNum = (int)(N * 0.02);
+        private const int CacheSize = (int)(N * 0.02);
+        private static volatile bool s_IsMatch;
+        private readonly string[] _regexes; // shuffled
+
+        public Perf_Regex_Cache()
+        {
+            _regexes = new string[N];
+            // create:
+            {
+                var i = 0;
+                for (; i < UniqueRegsNum; i++)
+                {
+                    // "(0+)" "(1+)" ..  "(9+)(9+)(8+)" ..
```

Comment:
Why these particular set of regexes? "(9+)(9+)(8+)" etc would not match "0123456789" and if you don't care whether there's a match you could have a simpler set of regexes eg "1", "2", ... "998" ...

Reply:
I think the pattern is fine as it may be used for others scenarios in the future as well